tonic/parameters.py

```diff
@@ -0,0 +1,158 @@
+"""
+    This is a generic Parameter class, and a ChildParameter class.
+
+    The Parameter class allows us to generate a parameter object,
+    with a certain value and a name. The name space is incremental
+    by defaul. Each Parameter object has a sympy symbol with its name.
+    
+    ChildParameter allows us to create dependent parameters 
+    with their values related to the parent Parameter value. This link
+    is given by an expression.
+
+
+    TODO:
+        1. Deal with displacement fields. 
+            These are symbolic objects generated by Geometric Entities.
+            If a displacement field is assosicated with a ChildParameter,
+            the parent Parameter should take care of the dependent disp-
+            lacement field(s).
+        2. Parameter attribute is_variable
+            I can see why we might need it, but perhaps the concept
+            should be more clear.
+"""
+
+import sympy
+
+
+class Parameter:
+    _parameters_counter = 1
+
+    def __init__(self, value, is_variable=True, name="p_"):
+        self.value = value
+        self.is_variable = is_variable
+
+        self.symbol = sympy.var(name + str(Parameter._parameters_counter))
+        Parameter._parameters_counter += 1
+
+        self.children = []
+
+    def __str__(self):
+        _parameter_info = {
+            "name": self.symbol,
+            "value": self.value,
+            "is_variable": self.is_variable,
+        }
+        return str(_parameter_info) + "\n"
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __add__(self, other):
+        """
+        Addition rules:
+            - Parameter + int / float -> ChildParameter
+                Adding number to a parameter does not introduce a new
+                parameter, but creates a connected (child) parameter.
+                When the parent's value changes, the child should also
+                change.
+            - Parameter + Parameter -> float
+                There is no point of creating a new Parameter
+                object when adding one variable parameters to another.
+                The construction is meaningless, but for debugging purpose
+                it is possible.
+        """
+        if isinstance(other, int) or isinstance(other, float):
+            return ChildParameter(self, self.symbol + other)
+        elif isinstance(other, Parameter) or isinstance(other, ChildParameter):
+            return self.value + other.value
+        else:
+            raise ValueError("int, float value or Parameter is required")
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    def __sub__(self, other):
+        if isinstance(other, int) or isinstance(other, float):
```

Comment:
I think its neater to overload `__neg__(self)`  for Parameters. That way, `__sub__` and `__rsub__` can be done by: `self + -other` and `other + -self`. This way you avoid having to repeat all this logic.